tests/test_host_port.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: Apache-2.0
+# pylint: disable=duplicate-code
+
+from click.testing import CliRunner
+from instructlab import lab
+
+
+class TestServe:
+    def test_serve_with_defaults(self):
```

Comment:
Use [pyest.mark.parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html) instead of repeat the same test code over and over again.